homework5/create_api_module.py

```diff
@@ -0,0 +1,316 @@
+import sqlite3
+import os
+import requests
+import csv
+from logger import setup_logger
+from db_decorator import db_connection_decorator
+import freecurrencyapi
+from datetime import datetime, timezone
+
+# from money_transfer import perform_money_transfer               cyclic import
+
+API_KEY = 'fca_live_4uqhXfyZENDdm83mAzfXYpguR4kCOXDt76l5cEIl'
+URL = f'https://api.freecurrencyapi.com/v1/latest?apikey={API_KEY}'
+
+my_logger = setup_logger()
+
+
+@db_connection_decorator
+def add_user(cursor, *input_data):
+    """
+        Add user information to the database.
+        Parameters:
+            cursor (sqlite3.Connection): The database connection.
+            input_data: User data containing dictionaries with keys 'user_full_name', 'birth_day', and 'accounts'.
+                Each dictionary represents a user to be added.
+        Note:
+            The 'user_full_name' is split into 'Name' and 'Surname' columns in the database.
+        Returns:
+            None
+        """
+    data = input_data[0] if isinstance(input_data[0], list) else input_data
+    # rows = [(user[0].split()[0], user[0].split()[1], user[1], user[2]) for user in data]
+    rows = [(row['user_full_name'].split()[0], row['user_full_name'].split()[1], row['birth_day'],
+             row['accounts']) for row in data]
+    cursor.executemany('INSERT INTO User (Name, Surname, Birth_day, Accounts) VALUES (?, ?, ?, ?)', rows)
+    my_logger.info('Users added successfully')
+
+
+@db_connection_decorator
+def add_bank(cursor, *input_data):
```

Comment:
all add functions look extremely similar. make another base function to reuse the code.